### PR TITLE
`create` directive

### PR DESCRIPTION
**Pull request by jesseleite**
Implementing a `create` directive (see issue #199 for details). Example...

```yaml
- create:
    - ~/path/to/some/cache/folder
    - ~/path/to/some/blank-log-file.txt
```

- [x] Create empty folders.
- ~~[ ] Create empty files.~~ (Not going to address in this PR)

### CODE REVIEW

File: dotbot/dispatcher.py [hunk 1]
```diff
@@ -27,8 +27,8 @@ def dispatch(self, tasks):
                     # keep going, let other plugins handle this if they want
                 for plugin in self._plugins:
                     if plugin.can_handle(action):
+                        success &= plugin.handle(action, task[action])
                         try:
-                            success &= plugin.handle(action, task[action])
```
Comment (jesseleite) [inline dotbot/dispatcher.py]: This is temporary while I'm debugging, didn't mean to commit this sorry.

**Comment by anishathalye** [comment]
Ooh nice, you've even written a test 😄 

See the discussion in #199. Perhaps we should only have creating directories, and skip the file creation part?

**Comment by jesseleite** [comment]
Btw, how about I clean this PR up for folder creation, and work on file creation in a separate PR?  It's kinda a different discussion, and that way you can critique it separately ☺️

**Comment by anishathalye** [comment]
Yes, that sounds good! I think folder creation is a good idea, but file creation might require more discussion.

**Comment by jesseleite** [comment]
Alright, think I'm done with this PR.  Take a look when you get a chance, no rush though ☺️

**Comment by anishathalye** [comment]
Thanks! From a quick skim, this looks good. I appreciate the attention to detail, like the change to the `shell` example. Thanks for the unit test as well!

I will try to take a closer look, and then squash and merge, in the next day or two.

**Comment by jesseleite** [comment]
Cool man, sounds good thank you ☺️

**Comment by anishathalye** [comment]
Squashed in 5a0f6676d41a7291e89f5e32e09396490d71a800 and merged in 04c113b5b85e87969fdbbab22d26bc4aa5398e69. Thanks again for your contribution!

**Pull request closed without merging**